lib/vm/src/instance.rs

```diff
@@ -688,6 +688,218 @@ impl Instance {
     }
 }
 
+/// TODO: rename this
+/// You must not put any type in this struct with drop logic as
+/// `Drop` will not be called on this type if it transitions into the
+/// next state.
+pub struct UnpreparedInstanceAllocatorSetter {
```

Comment:
What you think about `UnpreparedInstance` / `InstanceLayout` ?

Reply:
Well, it's certainly an improvement!

`UnpreparedInstance` isn't too bad, I'll review it with fresh eyes tomorrow